tensorflow_datasets/structured/iris.py

```diff
@@ -83,9 +90,13 @@ def _split_generators(self, dl_manager):
     ]
 
   def _generate_examples(self, records):
-    for record in records:
-      elems = record.split(",")
-      yield {
+    for i, row in enumerate(records):
+      elems = row.split(",")
+      record = {
           "features": [float(e) for e in elems[:-1]],
           "label": elems[-1],
       }
+      if self.version.implements(tfds.core.Experiment.S3):
+        yield i, record
+      else:
+        yield record
```

Comment:
you should add the `tfds.core.drop_key_if_not_s3` decorator to `_generate_examples`. The decorator added by pierrot, please check the last prs and add it.